dotCMS/src/main/java/com/dotcms/util/transform/DBTransformer.java

```diff
@@ -0,0 +1,18 @@
+package com.dotcms.util.transform;
+
+import com.dotmarketing.business.DotStateException;
+import java.util.List;
+
+/**
+ * Interface that contains the definition that a transformer needs to implement in order to convert
+ * DB objects into entities(eg. Folder, Template, Containers)
+ * @param <T>
+ */
+public interface DBTransformer<T> {
+
+    /**
+     * @return List of converted objects
+     * @throws DotStateException
+     */
+    List<T> asList() throws DotStateException;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'com.dotmarketing.business.DotStateException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)